packages/lib/gpc/src/gpcChecks.ts

```diff
@@ -0,0 +1,569 @@
+      // This named entry should exist in the given POD.
+      const podValue = pod.content.getValue(entryName);
+      if (podValue === undefined) {
+        throw new ReferenceError(
+          `Configured entry ${objName}.${entryName} doesn't exist in input.`
+        );
+      }
+
+      // If this entry identifies the owner, we should have a matching Identity.
+      if (entryConfig.isOwnerCommitment) {
+        hasOwnerEntry = true;
+        if (proofInputs.ownerSemaphoreV3 === undefined) {
+          throw new Error(
+            "Proof configuration expects owner, but no owner identity given."
+          );
+        }
+        if (podValue.value !== proofInputs.ownerSemaphoreV3.commitment) {
+          throw new Error(
+            `Configured owner commitment in POD doesn't match given Identity.`
+          );
+        }
+
+        // Owner commitment value must be a cryptographic number, so that its
+        // plaintext value can be included in the circuit.
+        if (podValue.type !== "cryptographic") {
+          throw new Error(
+            "Owner identity commitment must be of cryptographic type."
+          );
+        }
+      }
+
+      // Identified equal entry must also exist.
+      if (entryConfig.equalsEntry !== undefined) {
+        const { objName: otherPODName, entryName: otherEntryName } =
+          splitPODEntryIdentifier(entryConfig.equalsEntry);
+        const otherPOD = proofInputs.pods[otherPODName];
+        const otherValue = otherPOD?.content?.getValue(otherEntryName);
+        if (otherValue === undefined) {
+          throw new ReferenceError(
+            `Input entry ${objName}.${entryName} should be proved to equal` +
+              ` ${otherPODName}.${otherEntryName} which doesn't exist in input.`
+          );
+        }
+
+        if (podValueHash(otherValue) !== podValueHash(podValue)) {
+          throw new Error(
+            `Input entry ${objName}.${entryName} doesn't equal ${otherPODName}.${otherEntryName}.`
+          );
+        }
+      }
+    }
+  }
+
+  // Check nullifier configuration.
+  if (proofInputs.externalNullifier !== undefined && !hasOwnerEntry) {
+    throw new Error("Nullifier hash requires a defined owner entry.");
+  }
+}
+
+/**
+ * Checks the validity of the arguments for verifying a proof.  This will throw
+ * if any of the arguments is malformed, or if the different fields do not
+ * correctly correspond to each other.
+ *
+ * @param boundConfig proof configuration bound to a specific circuit
+ * @param revealedClaims revealed values from the proof
+ * @returns the minimum circuit parameter requirements for proving with
+ *   these arguments
+ * @throws TypeError if one of the objects is malformed
+ * @throws Error if logical requirements between fields are not met
+ */
+export function checkVerifyArgs(
+  boundConfig: GPCBoundConfig,
+  revealedClaims: GPCRevealedClaims
+): ProtoPODGPCCircuitParams {
```

Comment:
How/why so?  It needs to know the size of the modules in the circuits it uses somehow.  This compiler is tied specifically to a circuit family.

I do think that exposing these types outside of @pcd/gpc would be undesirable, which may be what you meant here, though I think that mostly because extracting circuit parameters is conceptually separate from checking, and only done here because it's convenient.  These checking functions aren't actually exported from the package, and I'd want to refactor a version without the return value if they were.  However the circuit description returned by gpcBindConfig is similarly tied to the ProtoPODGPC family.